src/interned.rs

```diff
@@ -605,6 +623,40 @@ where
             }),
         });
 
+        // Insert the newly allocated ID.
+        self.insert_id(id, zalsa, shard, hash, value);
+
+        let index = self.database_key_index(id);
```

Comment:
It's not clear to me how this change is related to persistant caching. Should this be a separate PR/fix?

Reply:
I just split out some of the functionality into the `insert_id` method so we can reuse it when deserializing, there should be no semantic change.